qcelemental/models/molecule.py

```diff
@@ -388,6 +389,11 @@ def orient_molecule(self):
         return Molecule(orient=True, **self.dict())
 
     def compare(self, other):
+        warnings.warn("Molecule.compare is deprecated and will be removed in v0.13.0. Use == instead.",
+                      DeprecationWarning)
```

Comment:
fwiw, this is the form we settled on for p4. `FutureWarning` https://github.com/psi4/psi4/blob/02bc2cbc17e8779b1cd6b0aa687760cee59e6fa8/psi4/driver/p4util/python_helpers.py#L711-L714